### PR TITLE
RUBY-3037: Support index wildcard_projection option

### PR DESCRIPTION
**Pull request by johnnyshields**
https://jira.mongodb.org/browse/RUBY-3037

### CODE REVIEW

**Comment by johnnyshields** [comment]
@comandeo yes this spec is still relevant.

I can't see the failure here. The specs are exiting with this line, so perhaps its something to do with MRSS?

```
rake aborted!
The following buckets failed: integration
/home/runner/work/mongo-ruby-driver/mongo-ruby-driver/spec/shared/lib/mrss/spec_organizer.rb:127:in `run_buckets'
/home/runner/work/mongo-ruby-driver/mongo-ruby-driver/spec/shared/lib/mrss/spec_organizer.rb:94:in `run'
/home/runner/work/mongo-ruby-driver/mongo-ruby-driver/Rakefile:100:in `block (2 levels) in <top (required)>'
/opt/hostedtoolcache/Ruby/2.5.9/x64/bin/bundle:23:in `load'
/opt/hostedtoolcache/Ruby/2.5.9/x64/bin/bundle:23:in `<main>'
Tasks: TOP => spec:ci
(See full trace by running task with --trace)
Error: Process completed with exit code 1.
```

**Comment by johnnyshields** [comment]
I've pulled in the latest master now to get the specs to re-run. Let's see what happens.

**Comment by p-mongo** [comment]
@johnnyshields Can you please review this PR and repair any failures in GH actions. If the failures are not related to this PR can you please make separate PR(s) as needed to repair the failures.

**Comment by johnnyshields** [comment]
@comandeo we should now be good on this one. Please run Evergreen, thanks!

File: spec/mongo/index/view_spec.rb [hunk 1]
```diff
@@ -909,6 +909,62 @@
       end
     end
 
+    context 'when providing an invalid wildcard projection expression' do
+      min_server_fcv '4.2'
+
+      it 'raises an exception' do
+        expect {
+          view.create_one({ '$**' => 1 }, wildcard_projection: 5)
+        }.to raise_error(Mongo::Error::OperationFailure)
```
[the remark this answers is not in the document]
Comment (johnnyshields) [reply spec/mongo/index/view_spec.rb]: Sure, will do. `rubocop-rspec` can warn for this case, FYI.

**Review by p-mongo — APPROVED** [review]

**Review by comandeo — APPROVED** [review]

**Review by Neilshweky — APPROVED** [review]

**Pull request merged**